dbms/src/Core/Spiller.h

```diff
@@ -45,23 +47,24 @@ struct SpilledFiles
 class Spiller
 {
 public:
-    Spiller(const String & id, bool is_input_sorted, size_t partition_num, const String & spill_dir, const Block & input_schema, const LoggerPtr & logger);
-    bool spillBlocks(const Blocks & blocks, size_t partition_id);
+    Spiller(const SpillConfig & config, bool is_input_sorted, size_t partition_num, const Block & input_schema, const LoggerPtr & logger);
+    void spillBlocks(const Blocks & blocks, size_t partition_id);
     BlockInputStreams restoreBlocks(size_t partition_id, size_t max_stream_size);
```

Comment:
```suggestion
    BlockInputStreams restoreBlocks(size_t partition_id, size_t max_stream_size = 0);
```
ref: `max_stream_size == 0 means the spiller choose the stream size automatically`